Automattic/wp-calypso
# Refactor concierge components to be reused for rescheduling

## Summary
This PR refactors some of the concierge components to make them more generic in order to be re-used to address gh-hg-495.
- The main wizard like component is updated to be more generic in order to accommodate the multi step for reschedule as well as the booking process
- The calendar page is extracted as a shared component to be used for both booking and rescheduling
- The confirmation page is extracted as a shared component to be used for booking, rescheduling and cancelling a concierge appointment
- The skeleton component will be now specific to the flow (booking, rescheduling)

## Testing
1. Update `export const WPCOM_CONCIERGE_SCHEDULE_ID = 1938;` in `client/me/concierge/constants.js` so that the internal calendar is used.
2. Access `http://calypso.localhost:3000/me/concierge/<business-site-slug>/book`
3. The skeleton should reflect the page layout
4. Add a message and continue to the calendar step
5. The calendar page should reflect the booking step ( header copy and button copy should suggest the booking action )
6. Book an appointment
7. Confirmation page copy should reflect a successful booking
8. Using a non business site slug should show the upsell component
